src/hotspot/share/opto/ifnode.cpp

```diff
@@ -700,16 +720,17 @@ const TypeInt* IfNode::filtered_int_type(PhaseGVN* gvn, Node* val, Node* if_proj
 //
 
 // Is the comparison for this If suitable for folding?
-bool IfNode::cmpi_folds(PhaseIterGVN* igvn, bool fold_ne) {
+bool IfNode::cmpi_folds(PhaseIterGVN* igvn) {
```

Comment:
We should probably rename it into `cmp_folds` now that it is also for unsigned and also update the description above. Shouldn't the description be at `fold_compares()` anyways? But maybe we want to clean this up in a follow-up RFE.

Reply:
Yes, will rename `cmpi_folds` to `cmp_folds`.

Not changing the fold compares description location for now.
For now maybe it is okay location for description as that is starting of fold_compares implementation (all methods above fold_compares definition seems mainly its helper functions)
.e.g.: See the comment and declarations in [cfgnode.hpp]
........
private:
  // Helper methods for fold_compares
  bool cmp_folds(PhaseIterGVN* igvn);
  bool is_ctrl_folds(Node* ctrl, PhaseIterGVN* igvn);
Or yes agree we can change this in a follow-up RFE.

Reply:
Yeah, it's probably best to leave it as it is at the moment and clean it up later in an RFE.